feat: add support for embeddings (#755)
envoyproxy/ai-gateway: site/docs/getting-started/connect-providers/openai.md

## Configuring More Models

To use more models, add more [AIGatewayRouteRule]s to the `basic.yaml` file with the [model alias] in the `value` field.

> For example, for chat completion models, to use [o1], for embedding models, to use [text-embedding-ada-002](https://platform.openai.com/docs/models/text-embedding-ada-002):

Reviewer: ```suggestion
For example, let's add [o1] as a chat completion model, and [text-embedding-ada-002](https://platform.openai.com/docs/models/text-embedding-ada-002) as an embedding model:
```

Author: Done

        - name: envoy-ai-gateway-basic-openai
```

> ### Embedding Models

Reviewer: can you delete this and simply add a new matching to AIGatewayRoute above
